pkg/common/azclient/trafficmanagerclient/custom.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+// Package trafficmanagerclient provides a client for Azure Traffic Manager profiles.
+// It wraps the Azure SDK for Go's Traffic Manager client and provides additional functionality
+// such as metrics collection.
+package trafficmanagerclient
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/trafficmanager/armtrafficmanager"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient/metrics"
+)
+
+const (
+	profileResourceType = "TrafficManagerProfile"
+)
+
+// ProfilesClient is a wrapper around the Azure Traffic Manager Profiles client.
+// It provides methods to interact with Traffic Manager profiles in Azure.
+type ProfilesClient struct {
+	*armtrafficmanager.ProfilesClient
+	subscriptionID string
+}
+
+// NewProfilesClient creates an Azure traffic manager profile client.
+func NewProfilesClient(client *armtrafficmanager.ProfilesClient, subscriptionID string) *ProfilesClient {
```

Comment:
where is this called? I am confused as I can't see who creates this metrics enabled client in this PR. What did I miss?